packages/vertica-nodejs/lib/backup-nodes-parsers.js

```diff
@@ -0,0 +1,168 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict'
+
+// Parser Combinators
+
+// pair the result of two parsers
+function pair(parser1, parser2) {
+  return input => {
+    var [nextInput, res1] = parser1(input)
+    var [finalInput, res2] = parser2(nextInput)
+    return [finalInput, [res1, res2]]
+  }
+}
+
+// helper function for right function
+function map(parser, fn) {
+  return input => {
+    var [nextInput, res] = parser(input)
+    return [nextInput, fn(res)]
+  }
+}
+
+// take result of right parser
+function right(parser1, parser2) {
+  return map(pair(parser1, parser2), t => t[1])
+}
+
+// try to parse input and if it fails, return null result
+function zeroOrOne(parser) {
+  return input => {
+    try {
+      return parser(input)
+    } catch(err) {
+      return [input, null]
+    }
+  }
+}
+
+// tries the first parser and if it fails, tries the second. If both fail, it throws an error.
+function either(parser1, parser2) {
+  return input => {
+    try {
+      return parser1(input)
+    } catch(e1) {
+      try {
+        return parser2(input)
+      } catch(e2) {
+        throw new Error("Unable to parse input with either parser provided:\nparser 1 error: (" + e1 + ")\nparser 2 error: (" + e2 + ")")
+      }
+    }
+  }
+}
+
+
+// Parsers
+
+// Parse an IPv4 address by checking if the input contains a . in it
+function ipv4Address(input) {
+  var result = ""  
+    
+  for (let c of input) {
+    if (c === ':') {
+      break
+    }
+    result += c
+  }
+    
+  if (result.includes('.')) {
+    return [input.substring(result.length), result]
+  } else {
+    throw new Error("Unable to parse " + input + " as an IPv4 address")
+  }
+}
+
+// Parse an IPv6 address by checking if it's wrapped by [] or contains multiple colons
+function ipv6Address(input) {
+  var result = ""
+  if (input.startsWith('[')) {
+    // use substring to cut off the bracket first
+    for (let c of input.substring(1)) {
+      if (c === ']') {
+        break
+      }
+      result += c
+    }
+    // add one to cut off the end bracket from the next input
+    return [input.substring(result.length + 2), result]
+  }
+  
+  var colonCounter = 0
+  for (let c of input) {
+    if (c === ':') {
+      colonCounter++
+    }
+
+    result += c
+  }
+
+  if (colonCounter > 1) {
+    return [input.substring(result.length), result]
+  }
+
+  throw new Error("Unable to parse " + input + " as an IPv6 address")
+}
+
+// Parses a host name
+function host(input) {
+  var hostname = ""
+    
+  for (let c of input) {
+    if (c === ':') {
+      break
+    }
+    hostname += c
+  }
+   
+  return [input.substring(hostname.length), hostname]
```

Comment:
Does this distinguish between hostname and IP addresses?

Reply:
As mentioned in the other comment, we do not really distinguish between hostnames and IP addresses properly, but the current logic happens to work for the cases we care about. More details in the other comment.